module-check/src/main/clojure/clojure/core/typed/base_env.clj

```diff
@@ -1443,7 +1443,17 @@ clojure.core/rand-nth (All [x] [(U (Indexed x) (SequentialSeqable x)) -> x])
   this-ns
 clojure.set/union (All [x] [(Set x) * -> (Set x)])
 clojure.set/intersection (All [x] [(Set x) (Set x) * -> (Set x)])
+
 clojure.set/difference (All [x] [(Set x) (Set Any) * -> (Set x)])
+clojure.set/subset? [(Set Any) (Set Any) -> Boolean]
+clojure.set/superset? [(Set Any) (Set Any) -> Boolean]
+clojure.set/join [(Set (Map Any Any)) (Set (Map Any Any)) -> (Set (Map Any Any))]
+clojure.set/index [(Set (Map Any Any)) (Vec Any) -> (Map (Map Any Any) (Set (Map Any Any)))]
+clojure.set/map-invert (All [a b] [(Map a b) -> (Map b a)])
+clojure.set/project [(Set (Map Any Any)) (Vec Any) -> (Set (Map Any Any))]
+clojure.set/rename [(Set Any) (Map Any Any) -> (Set Any)]
+clojure.set/rename-keys [(Map Any Any) (Map Any Any) -> (Map Any Any)]
+clojure.set/select [ (IFn [AnyInteger -> java.lang.Boolean]) (Set Any) -> (Set Any)]
```

Comment:
I'd rather be conservative in `clojure.set`. Based on the argument name being `xset` it seems it's intended to be a set.

I'd prefer

```clojure
(All [a] [[a -> Boolean] (Set a) -> (Set a)])
```

We could go a step further and copy the type of `filter` to recognise the output only has the types the predicate passes, but I'm fine with this type.